### PR TITLE
Metric tests

### PR DESCRIPTION
**Pull request by isabelcabezasm**
This pull request adds tests for the **entity extraction**, **accuracy** for entities and  **topic coverage** evaluation features in the `QAEvalEngine`. 

It introduces new test modules to validate both "unit tests" and "integration scenarios" (calling LLM in azure), ensuring the correctness of entity extraction, topic coverage scoring, and prompt formatting for different edge cases and realistic examples.

## Summary by CodeRabbit

* **New Features**
  * Expanded entity-extraction guidance to five explicit categories with examples and clearer instructions.
  * Refined topic-coverage evaluation flow with explicit empty-list handling, standardized match scoring, and clearer coverage calculation.

* **Tests**
  * Added comprehensive unit and integration tests and shared test utilities for accuracy evaluation, entity extraction, and topic-coverage, including sample data and edge-case coverage.

### CODE REVIEW

File: tests/eval/test_accuracy.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+
+import pytest
+from tests.eval.common import (
+    assert_mock_agent_called_correctly,
+    mock_engine,  # pyright: ignore[reportUnusedImport] it's a fixture
+    parse_entity_string,
+    requires_azure,
+    sample_entity_extraction_result,
+    sample_entity_extraction_with_overlap,
+)
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply tests/eval/test_accuracy.py]: false positive, it's used, it's a fixture. used in line 59

File: src/eval/llm_evaluator/prompts/topic_coverage_prompt.md [hunk 2]
```diff
@@ -4,15 +4,18 @@ Generated entities: {generated_entities}
...
 Your task is to:
 
-1. Check the entities of both list, consider synonyms or equivalent terms (e.g.,
-"interest rate" is equivalent to "policy rate", or "inflation" is equivalent to "price increases"). Add those considerations in the reason property. 
+1. If either list is empty:
+   - State clearly which list is empty.
+   - Set coverage score to 1 if both lists are empty (since there’s nothing expected and nothing generated).
+   - Set coverage score to 0 if expected_entities is not empty and generated_entities is empty.
+   - Provide a reason explaining the situation (e.g., "No entities were expected" or "No entities were generated").
+   - Do NOT attempt synonym matching in this case.
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/llm_evaluator/prompts/topic_coverage_prompt.md]: solved: 2ad69a2660b4e468c06c36b1f611496bcee751d9

File: tests/eval/test_topic_coverage.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    )
+
+    assert isinstance(result, TopicCoverageEvaluationResults)
+    assert math.isclose(
+        result.coverage_score, expected_score, rel_tol=1e-2, abs_tol=1e-2
+    )
+
+    # Verify the mock agent was called exactly once with:
+    # - Correct response_format (EntityExtraction)
+    # - All input texts (user_query, llm_answer, expected_answer) in the
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply tests/eval/test_topic_coverage.py]: fixed here: db831d3a99af05efad060a4df1b3e71a6de3f6a6
